glirc: unbreak via downgrade to 2.38

aforemny opened the pull request:
glirc 2.39 does not compile against any set of released packages and is unlikely to be fixed without a subsequent release (https://github.com/glguy/irc-core/issues/99).

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

sternenseemann commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -2550,8 +2550,8 @@ self: super: {
-  glirc = assert super.irc-core.version == "2.10"; doJailbreak super.glirc;
-  hookup = assert super.irc-core.version == "2.10"; doJailbreak super.hookup;
+  # glirc 2.39 is broken, downgraded to 2.38 to unbreak
+  glirc = assert super.irc-core.version == "2.10"; doJailbreak (markUnbroken super.glirc);
```
Just delete `glirc` from `broken.yaml` instead and regenerate the package set.

aforemny replied:
I tried that first, but that still pulled in glirc 2.39 and marked it as not broken. Is that still preferable?

aforemny replied:
I've pushed a commit that removes glirc from `broken.yaml`. Now `haskellPackages.glirc_2_39` is not marked as broken, but fails to build. Should I mark this broken in `configuration-common.nix` then?

sternenseemann replied:
Right, you need to add the default-package-overrides as well of course.

sternenseemann replied:
Oh that was what you meant! Don't bother, we don't give any guarantees over the versioned attributes (they are not built usually), so it's not required to mark them correctly!

sternenseemann reviewed (commented):
Can you rebase on haskell-updates?

aforemny committed:
glirc: unbreak via downgrade to 2.38

glirc 2.39 does not compile against any set of released packages and is
unlikely to be fixed without a subsequent release
(https://github.com/glguy/irc-core/issues/99).

sternenseemann commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -2550,8 +2550,8 @@ self: super: {
-  glirc = assert super.irc-core.version == "2.10"; doJailbreak super.glirc;
-  hookup = assert super.irc-core.version == "2.10"; doJailbreak super.hookup;
+  # glirc 2.39 is broken, downgraded to 2.38 to unbreak
+  glirc = assert super.irc-core.version == "2.10"; doJailbreak (markUnbroken super.glirc);
```
`markUnbroken` can be dropped now. Not sure if the jailbreaks are still necessary (probably?).

aforemny replied:
Right! Jailbreaking is still necessary for attoparsec bounds present in glirc 2.38 and irc-core.

sternenseemann commented:
@ofborg build haskellPackages.glirc

aforemny commented:
Resolved comments, squashed and rebased onto haskell-updates.

And accidentally (how?) requested a review from a bunch of additional people. Yikes, my apologies!
